### ISSUE TITLE
ApplicationInsights doesn't set x-ms-request-root-id & x-ms-request-id headers in Internet Explorer 11

### ISSUE THREAD

**Issue by thebitllc** [Microsoft/ApplicationInsights-JS#414]
I was testing Application Insights with IE and noticed that the request headers weren't getting set in the following code:

```
                if (!this.appInsights.config.disableCorrelationHeaders && (ApplicationInsights.UrlHelper.parseUrl(xhr.ajaxData.getAbsoluteUrl()).host == this.currentWindowHost)) {
                    var rootId = this.appInsights.context.operation.id;
                    xhr.setRequestHeader("x-ms-request-root-id", rootId);
                    xhr.setRequestHeader("x-ms-request-id", xhr.ajaxData.id);
                }
```

ApplicationInsights.UrlHelper.parseUrl(xhr.ajaxData.getAbsoluteUrl()).host == this.currentWindowHost) evaluates to false. Because ApplicationInsights.UrlHelper.parseUrl(xhr.ajaxData.getAbsoluteUrl()).host returns a port number at the end, like:

subdomain.domain.com:443 while this.currentWindowHost is subdomain.domain.com

**Comment by emasyakin-softheme** [comment]
Imagine a use-case:
static html/js app at app.example.com
separate server hosted at api.example.com

I want api.domain.com to know about request-id from app.domain.com to correlate.

Looks like it`s not currently possible.

**Comment by thebitllc** [comment]
I don't believe this issue was fixed, I am testing 1.0.20. I have exactly the same problem.
